### PR TITLE
Review casts in PCL, Part A [#4230]

### PR DESCRIPTION
**Pull request by gnawme**
To address issue #4230, this PR:

- adds [`cppcoreguidelines-pro-type-cstyle-cast`](https://releases.llvm.org/13.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/cppcoreguidelines-pro-type-cstyle-cast.html) to `.clang-tidy` to fix the use of C-style casts to cast unrelated types
- adds [`cppcoreguidelines-pro-type-static-cast-downcast`](https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines/pro-type-static-cast-downcast.html) to `.clang-tidy` to fix the use of `static_cast` where `dynamic_cast` should have been used instead.

(@mvieth Please assign #4230 to me.)

### CODE REVIEW

**Review by mvieth — COMMENTED** [review]
I can assign the issue to you if you want, but I think it is not asking for the kind of changes you make in this pull request (thank you for this pull request nonetheless)

File: io/src/robot_eye_grabber.cpp [hunk 1]
```diff
@@ -176,8 +178,10 @@ pcl::RobotEyeGrabber::convertPacketData (unsigned char *data_packet, std::size_t
     //The new packet data format contains this as a header
     //char[6]  "EBRBEP"
     //std::uint32_t Timestamp // counts of a 66 MHz clock since power-on of eye.
+    const std::string PACKET_HEADER("EBRBEP");
+    std::string packet(reinterpret_cast<const char*>(data_packet));
```
[the remark this answers is not in the document]
Comment (gnawme) [reply io/src/robot_eye_grabber.cpp]: Changed constructor to use the first six values of `data_packet`

**Comment by gnawme** [comment]
> I can assign the issue to you if you want, but I think it is not asking for the kind of changes you make in this pull request (thank you for this pull request nonetheless)

Ah; Part B may be closer to the intent of #4230, as it converts many C-style casts to C++, and removes a few redundant casts as well. But I guess I don't necessarily need an issue to open a PR...

**Review by mvieth — APPROVED** [review]
Thanks!

**Pull request merged**